src/System.IO.Pipelines/System/IO/Pipelines/PreservedBuffer.cs

```diff
@@ -25,23 +25,22 @@ internal PreservedBuffer(ref ReadableBuffer buffer)
         /// </summary>
         public void Dispose()
         {
-            var returnStart = _buffer.Start.Segment;
-            var returnEnd = _buffer.End.Segment;
-
-            while (true)
+            if (_buffer.Start.Segment is BufferSegment returnStart)
             {
-                var returnSegment = returnStart;
-                returnStart = returnStart?.Next;
-                returnSegment?.ResetMemory();
+                var returnEnd = _buffer.End.GetSegment();
 
-                if (returnSegment == returnEnd)
+                while (true)
                 {
-                    break;
+                    var returnSegment = returnStart;
+                    returnStart = returnStart?.Next;
+                    returnSegment?.ResetMemory();
+
+                    if (returnSegment == returnEnd)
+                    {
+                        break;
+                    }
                 }
             }
-
-            _buffer = default;
```

Comment:
Why remove this?

Reply:
It's clearing the field inside a specific copy of a struct which doesn't guarantee anything

Reply:
Of course not. I figured it was about referencing PreservedBuffer.Buffer after disposal. Better to nullref than allow it.

I wouldn't have mentioned it if it wasn't there before, but since you went out of your way to remove it.